### PR TITLE
CV3-50 Add book calendar event functionality

### PR DESCRIPTION
**Pull request by adafia**
#### Description
Given an available room, start time, end time and other user information, the Backend should be able to book a meeting that will appear on the Google calendar.

#### Type of change
Please select the relevant option

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Others (cosmetics, styling, improvements)
- [ ] This change requires a documentation update

#### How Has This Been Tested?

- [x] Unit
- [x] Integration
- [ ] End-to-end

#### How to Test

- Follow the instruction in the `readme.md` to get the project setup locally and run the 
   following GraphQL mutation on this `URL: ` `http://localhost:8000/mrm` 
   using the `HTTP: POST` verb. 
- Mutation
```
mutation {
    bookEvent(
        eventTitle:"Hello Kigali",
        startDate:"2019-11-16",
        startTime:"10:00",
        duration:60
        attendees: "adafia.samuel@gmail.com, samuel.adafia@andela.com",
        timeZone: "Africa/Kigali",
        room:"Yankari"
    ){
            response
        }
    }
```
- Expected response 
```
{
  "data": {
    "bookEvent": {
      "response": "Event created successfully"
    }
  }
}
```

#### JIRA
[CV3-50](https://andela-apprenticeship.atlassian.net/browse/CV3-50)

### CODE REVIEW

File: api/events/schema.py [hunk 1]
```diff
@@ -35,6 +39,80 @@ class Meta:
...
+
+    class Arguments:
+        event_title = graphene.String(required=True)
+        start_date = graphene.String(required=True)
+        start_time = graphene.String(required=True)
+        duration = graphene.Float(required=False)
+        attendees = graphene.String(required=False)
+        organizer = graphene.String(required=False)
+        description = graphene.String(required=False)
+        room = graphene.String(required=False)
```
Comment (mifeille) [inline api/events/schema.py]: I think the room should be required since you are booking a room

Comment (adafia) [reply api/events/schema.py]: Thank you @mifeille, I'll make the change and revert back to you.

File: api/events/schema.py [hunk 2]
```diff
@@ -35,6 +39,80 @@ class Meta:
...
+
+    @Auth.user_roles('Admin', 'Default User', 'Super Admin')
+    def mutate(self, info, **kwargs):
+        """Creates calendar events
+
+        Args:
+            event_title: A sting that communicates the event summary/title [required]
+            start_date: The start date of the event eg. 'Nov 4 2019' [required]
+            start_time: The end time of the event eg '07:00 AM' [required]
+            duration: A float of the duration of the event
```
Comment (mifeille) [inline api/events/schema.py]: I can not see the event end time, does it mean that the end time will be generated using this 'duration field'?

Comment (adafia) [reply api/events/schema.py]: Yes @mifeille, I created a helper that computes the event end time by adding the event duration to the start time provided by the user.

**Review by mifeille — COMMENTED** [review]
I think you have to validate the required fields cause I tried to create an event with an empty title and it worked, same for dates.

**Review by mifeille — APPROVED** [review]

**Review by MCFrank16 — APPROVED** [review]

**Review by joshuaocero — APPROVED** [review]

**Pull request merged**